codex/node.nim

```diff
@@ -327,13 +310,25 @@ proc start*(node: CodexNodeRef) {.async.} =
     # TODO: remove Sales callbacks, pass BlockStore and StorageProofs instead
     contracts.sales.onStore = proc(cid: string, _: Availability) {.async.} =
       # store data in local storage
-      (await node.store(Cid.init(cid).tryGet())).tryGet()
+      without cid =? Cid.init(cid):
+        trace "Unable to parse Cid", cid
+        return
```

Comment:
The onStore callback needs to fail with an exception here, otherwise the sales agent will happily go along with trying to fill a slot, even though it doesn't have the data to generate a proof.